mmv1/products/databasemigrationservice/connectionprofile.yaml

```diff
@@ -0,0 +1,499 @@
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'ConnectionProfile'
+self_link: 'projects/{{project}}/locations/{{location}}/connectionProfiles/{{connection_profile_id}}'
+base_url: 'projects/{{project}}/locations/{{location}}/connectionProfiles'
+create_url: 'projects/{{project}}/locations/{{location}}/connectionProfiles?connectionProfileId={{connection_profile_id}}'
+update_verb: :PATCH 
+update_mask: true
+description: 'A connection profile definition.'
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Database Migration': 'https://cloud.google.com/database-migration/docs/'
+  api: 'https://cloud.google.com/database-migration/docs/reference/rest/v1/projects.locations.connectionProfiles/create'
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'connectionProfileId'
+    required: true
+    input: true
+    url_param_only: true
+    description: |
+      The ID of the connection profile.
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    input: true
+    url_param_only: true
+    description: |
+      The location where the connection profile should reside.
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    description: |
+      The name of this connection profile resource in the form of projects/{project}/locations/{location}/connectionProfiles/{connectionProfile}.
+  - !ruby/object:Api::Type::String
+    name: 'displayName'
+    description: |
+      The connection profile display name.      
+  - !ruby/object:Api::Type::Time
+    name: 'createTime'
+    output: true
+    description: |
+      Output only. The timestamp when the resource was created. A timestamp in RFC3339 UTC 'Zulu' format, accurate to nanoseconds. Example: '2014-10-02T15:01:23.045123456Z'.
+  - !ruby/object:Api::Type::Time
+    name: 'updateime'
+    output: true
+    description: |
+      Output only. The timestamp when the resource was last updated. A timestamp in RFC3339 UTC 'Zulu' format, accurate to nanoseconds. Example: '2014-10-02T15:01:23.045123456Z'.
+  - !ruby/object:Api::Type::KeyValuePairs
+    name: 'labels'
+    description: |
+      The resource labels for connection profile to use to annotate any related underlying resources such as Compute Engine VMs.
+  - !ruby/object:Api::Type::Enum
+    name: 'state'
+    description: |
+      The current connection profile state.
+    values:
+      - :DRAFT
+      - :READY
+      - :FAILED
```

Comment:
Can this value be set (?) . Due the value enumerations my intuition leads me to believe it can not. The would mean it being either 
1. omitted as it's a transient field (like update_time)
2. set to output only

Reply:
I'm going to go with output only since it makes sense for the user to see which state the connection profile is in without opening up cloud console or making some separate request, as it affects what other changes can be made to the resource state.